src/webgpu/util/color_space_conversion.ts

```diff
@@ -230,11 +230,10 @@ export function makeInPlaceColorConversion({
     // This technically represents colors outside the src gamut, so no clamping yet.
 
     if (requireColorSpaceConversion) {
-      // WebGPU currently only supports dstColorSpace = 'srgb'.
       if (srcColorSpace === 'display-p3' && dstColorSpace === 'srgb') {
-        rgba = displayP3ToSrgb(rgba);
+        Object.assign(rgba, displayP3ToSrgb(rgba));
       } else {
-        unreachable();
+        Object.assign(rgba, srgbToDisplayP3(rgba));
       }
```

Comment:
... (GitHub suggestion feature being a pain about suggesting multiple lines.)
```suggestion
      } else {
        unreachable();
      }
```